operator/validator/controller.go

```diff
@@ -543,29 +533,6 @@ func (c *controller) StartValidators() {
 		// Start validators.
 		c.startValidators(inited, committees)
 	}
-
-	// Fetch metadata now if there is none. Otherwise, UpdateValidatorsMetadataLoop will handle it.
-	var hasMetadata bool
-	for _, share := range shares {
-		if !share.Liquidated && share.HasBeaconMetadata() {
-			hasMetadata = true
-			break
-		}
-	}
-	if !hasMetadata {
```

Comment:
I'm not sure why but the original logic used to fetch and update if none of the shares have metadata (the `!share.Liquidated` part of the condition is useless because `shares` contains non-liquidated shares)

This is changed to fetching all public keys if at least one share metadata is missing, however, probably we want to modify this logic

Reply:
As the comment says, initial fetch was only if we didn't have any metadata. because it relayed on the periodic metadata loop. In this PR we WANT to achieve an initial metadata sync to all relevant validators (before p2p starts)